grails-app/controllers/streama/UserController.groovy

```diff
@@ -119,6 +122,37 @@ class UserController {
     respond userInstance, [status: CREATED]
   }
 
+  @Transactional
+  def saveAndCreateUser() {
```

Comment:
1. This method is very much like the `saveAndInviteUser`. Code duplication should be avoided([DRY](https://en.wikipedia.org/wiki/Don't_repeat_yourself)).
2. And it does more than one thing([SRP](https://en.wikipedia.org/wiki/Single_responsibility_principle)). Its name says that.

I think that you can break both `saveAndInviteUser ` and `saveAndCreateUser` methods into smaller pieces while eliminating duplicated lines. Write separate methods that _creates and validates the user_, _saves the user and role in database_ and _sends email_.

This pseudocode will give you an idea on how to start
```
def createAccountWithEmail() {
  User u = createUser()
  if (u==null) {
    return
  }
  createRole(u)
  sendEmail(u)
  u.save
}
``` 

@dularion what are your thought on this?